src/shared/components/palette/base-field/abstract-form-field.component.ts

```diff
@@ -0,0 +1,34 @@
+import { Input } from '@angular/core';
+import { CaseField } from '../../../domain/definition';
+import { AbstractControl, FormControl, FormGroup } from '@angular/forms';
+
+export abstract class AbstractFormFieldComponent {
+
+  @Input()
+  caseField: CaseField;
+
+  @Input()
+  formGroup?: FormGroup;
+
+  @Input()
+  registerControl?: <T extends AbstractControl> (control: T) => T;
+
+  protected defaultControlRegister(): (control: FormControl) => AbstractControl {
+    return control => {
+      if (!this.formGroup) {
+        return null;
+      }
+      if (this.formGroup.controls[this.caseField.id]) {
+        return this.formGroup.get(this.caseField.id);
+      }
+      this.addValidators(this.caseField, control);
+      this.formGroup.addControl(this.caseField.id, control);
+      return control;
+    };
+  }
+
+  protected addValidators(caseField: CaseField, control: FormControl): void {
+    // No validators by default, override this method to add validators to the form control
```

Comment:
only write field require validation am I correct? In that case I was thinking whether we could remove the addValidators logic from here. And leave it only in the write field subclass. Also removing 
`this.addValidators(this.caseField, control);` 
from the default control register. The write field subclass could create a custom control register that calls the default and then register validators to the validator service

Reply:
defaultControlRegister returns AbstractControl , so just wanted to avoid downcasting AbstractControl to FormControl in field write component as FormValidatorsService expects FormControl. Do you still want me to do that or leave it as it is?